templates/CRM/UF/Form/AdvanceSetting.tpl

```diff
@@ -102,3 +107,24 @@
     </div><!-- / .crm-block -->
   </div><!-- /.crm-accordion-body -->
 </div><!-- /.crm-accordion-wrapper -->
+
+<script type="text/javascript">
+    {literal}
+    CRM.$(function () {
+        CRM.$("[name=is_update_dupe]").each(function () {
+            this.addEventListener('change', confirmCheck);
+            if (CRM.$(this).attr('checked'))
+                this.dispatchEvent(new Event('change'));
+        });
+    });
+
+    function confirmCheck() {
+        var duplicateURLInput = CRM.$("#for_duplicate_URL_block");
+        if (CRM.$(this).attr('checked') && CRM.$(this).val() == 1) {
+            duplicateURLInput.css('display', '');
+        } else if (duplicateURLInput.css('display') !== 'none') {
+            duplicateURLInput.css('display', 'none');
+        }
+    }
+    {/literal}
```

Comment:
This looks much better. A couple more cleanup tasks:

- Add `$` to the closure function argument so it looks like `CRM.$(function($) {`
- You can now use the shorthand `$` for jQuery inside the closure instead of `CRM.$`
- Move the function `confirmCheck` inside the closure and use `$` instead of `CRM.$` in that function.
- Use jquery function `.change()` instead of `addEventListener` and`dispatchEvent`.
- In general I don't like the practice of firing `change` on something that isn't actually changing because it can confuse things. I would rewrite that as `$("[name=is_update_dupe]").change(confirmCheck).filter(':checked').each(confirmCheck);` which in one line of code will attach the change handler and also immediately fire it for all checked boxes.

Reply:
Hi colemanw,
Sorry for delay response.

This code change is required in order to notify user that he is already subscribed.
Hence he will be redirected to the $duplicateUrl(some page with appropriate information) rather than $postUrl(misleading step in our case), like it was before.